zerver/lib/actions.py

```diff
@@ -3433,7 +3433,7 @@ def do_change_default_sending_stream(user_profile: UserProfile, stream: Optional
         event_time=event_time, modified_user=user_profile,
         acting_user=acting_user, extra_data=ujson.dumps({
             RealmAuditLog.OLD_VALUE: old_value,
-            RealmAuditLog.NEW_VALUE: stream,
+            RealmAuditLog.NEW_VALUE: None if stream is None else stream.id,
```

Comment:
Serializing the `stream` object was crazy, but I’m not sure what the intention was here, so I guessed. Also does this need a migration?

Reply:
What did that serialize as?  This doesn't need an immediate migration (since it's be quite clear which entries were written this way from not being str(integer)s), but we likely will want some sort of cleanup migration (if it's possible) so that we can rely on these RealmAuditLog entries in future migrations.  This can be handled as a separate issue I think.

Reply:
> What did that serialize as?

[A gigantic block of garbage](https://github.com/zulip/zulip/pull/15601#pullrequestreview-462970230).

Reply:
https://github.com/zulip/zulip/issues/16066 will resolve the follow-up issue.